upstream/upstream_test.go

```diff
@@ -246,38 +223,228 @@ func (suite *ListSuite) TestReconcile() {
 	suite.Assert().Equal(mockBackend("six"), backend)
 	suite.Assert().NoError(err)
 
-	l.Down(mockBackend("four")) // score == 2
-	l.Down(mockBackend("four")) // score == 1
-	l.Down(mockBackend("four")) // score == 0
-	l.Down(mockBackend("four")) // score == -1
+	l.Down("four") // score == 2
+	l.Down("four") // score == 1
+	l.Down("four") // score == 0
+	l.Down("four") // score == -1
 
 	backend, err = l.Pick()
-	suite.Assert().Equal(mockBackend("five"), backend)
+	// In previus version of the code, this would have been "five" but with additional scoring
+	// we sort the slice but do not touch the current index, so we get "six" instead.
+	suite.Assert().Equal(mockBackend("six"), backend)
 	suite.Assert().NoError(err)
 
 	backend, err = l.Pick()
-	suite.Assert().Equal(mockBackend("six"), backend)
+	suite.Assert().Equal(mockBackend("five"), backend)
 	suite.Assert().NoError(err)
 
-	l.Reconcile([]upstream.Backend{
-		mockBackend("five"),
-		mockBackend("six"),
-		mockBackend("four"),
-	})
+	l.Reconcile([]mockBackend{"five", "six", "four"})
 
 	backend, err = l.Pick()
-	suite.Assert().Equal(mockBackend("five"), backend)
+	suite.Assert().Equal(mockBackend("six"), backend)
 	suite.Assert().NoError(err)
 
 	backend, err = l.Pick()
-	suite.Assert().Equal(mockBackend("six"), backend)
+	suite.Assert().Equal(mockBackend("five"), backend)
 	suite.Assert().NoError(err)
 
 	l.Reconcile(nil)
 
 	backend, err = l.Pick()
-	suite.Assert().Nil(backend)
+	suite.Assert().Zero(backend)
+	suite.Assert().EqualError(err, "no upstreams available")
+}
+
+func (suite *ListSuite) TestBalancing() {
+	l, lerr := upstream.NewList(
+		[]mockBackend{"one", "two", "three"},
+		upstream.WithLowHighScores(-3, 3),
+		upstream.WithInitialScore(1),
+		upstream.WithScoreDeltas(-1, 1),
+		upstream.WithHealthcheckInterval(time.Hour),
+		upstream.WithHealthcheckTimeout(time.Hour),
+	)
+	suite.Require().NoError(lerr)
+
+	defer l.Shutdown()
+
+	// Move one to sub millisecond range
```

Comment:
```suggestion
	// Move one to sub microsecond range
```

Reply:
ignore, since  submillisecond means almost a 1000 microseconds, :sweat_smile:

Reply:
Yeah I just decided to reword it completely an unify with other comments.